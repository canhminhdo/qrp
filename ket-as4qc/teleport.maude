load qc.maude

fmod TELEPORT is
    pr QC .

    ops a b : -> Cpx .
    op |psi> : -> Vect .

    eq |psi> = a . |0> + b . |1> .
endfm

red (P0(1) x P0(0) x H(0) x CX(0, 1) x CX(1,2) x H(1)) x (|psi> (x) |0> (x) |0>) .
red (Z(2) x P0(1) x P1(0) x H(0) x CX(0, 1) x CX(1,2) x H(1)) x (|psi> (x) |0> (x) |0>) .
red (X(2) x P1(1) x P0(0) x H(0) x CX(0, 1) x CX(1,2) x H(1)) x (|psi> (x) |0> (x) |0>) .
red (Z(2) x X(2) x P1(1) x P1(0) x H(0) x CX(0, 1) x CX(1,2) x H(1)) x (|psi> (x) |0> (x) |0>) .
