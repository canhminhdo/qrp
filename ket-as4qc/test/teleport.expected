==========================================
reduce in TELEPORT : (P0(1) x P0(0) x H(0) x CX(0, 1) x CX(1, 2) x H(1)) x
    |psi>(x)|0>(x)|0> .
result Vect: (a .* 1/2) . |0>(x)|0>(x)|0> + (b .* 1/2) . |0>(x)|0>(x)|1>
==========================================
reduce in TELEPORT : (Z(2) x P0(1) x P1(0) x H(0) x CX(0, 1) x CX(1, 2) x H(1))
    x |psi>(x)|0>(x)|0> .
result Vect: (a .* 1/2) . |1>(x)|0>(x)|0> + (b .* 1/2) . |1>(x)|0>(x)|1>
==========================================
reduce in TELEPORT : (X(2) x P1(1) x P0(0) x H(0) x CX(0, 1) x CX(1, 2) x H(1))
    x |psi>(x)|0>(x)|0> .
result Vect: (a .* 1/2) . |0>(x)|1>(x)|0> + (b .* 1/2) . |0>(x)|1>(x)|1>
==========================================
reduce in TELEPORT : (Z(2) x X(2) x P1(1) x P1(0) x H(0) x CX(0, 1) x CX(1, 2)
    x H(1)) x |psi>(x)|0>(x)|0> .
result Vect: (a .* 1/2) . |1>(x)|1>(x)|0> + (b .* 1/2) . |1>(x)|1>(x)|1>
Bye.
